### PR TITLE
Converts a number of simple_animals into their simple_mob variants

### PR DESCRIPTION
**Pull request by Anewbe**
Converts every instance of the simple_animal mobs into its simple_mob equivalent, where possible. This includes the main map and the PoIs.

Also brings the branch up to date.

### CODE REVIEW

**Comment by Atermonera** [comment]
CaveTrench: `maps/submaps/surface_submaps/mountains/CaveTrench.dmm:12:error: undefined type: /mob/living/simple_animal/fish/trout`
RationCache: Two instances of the `returns_home` var, which is apparently invalid
BoomBase: One instance of 'returns_home'
DoomP: One instance of 'returns_home'

**Comment by Anewbe** [comment]
Sweet merciful RNGsus, why is this so awful?

**Comment by Anewbe** [comment]
- code/game/gamemodes/cult/cult_structures.dm
- code/game/gamemodes/technomancer/assistance/golem.dm
- code/game/gamemodes/technomancer/spells/control.dm
- code/game/gamemodes/technomancer/spells/summon/summon_creature.dm
- code/game/objects/effects/spiders.dm
- code/game/objects/items/glassjar.dm
- code/game/objects/items/weapons/grenades/spawnergrenade.dm
- code/game/objects/random/mob.dm
- code/game/turfs/simulated/water.dm
- code/modules/admin/topic.dm
- code/modules/events/infestation.dm
- code/modules/events/rogue_drones.dm
- code/modules/hydroponics/seed.dm
- code/modules/mob/_modifiers/traits_phobias.dm
- code/modules/mob/living/silicon/robot/drone/drone_items.dm
- code/modules/mob/living/simple_mob/subtypes/animal/giant_spider/carrier.dm
- code/modules/mob/living/simple_mob/subtypes/animal/giant_spider/nurse.dm
- code/modules/mob/living/simple_mob/subtypes/animal/giant_spider/webslinger.dm
- code/modules/mob/living/simple_mob/subtypes/humanoid/mercs/mercs.dm
- code/modules/mob/living/simple_mob/subtypes/mechanical/hivebot/hivebot.dm
- code/modules/mob/living/simple_mob/subtypes/mechanical/hivebot/ranged_damage.dm
- code/modules/mob/mob_movement.dm
- code/modules/mob/transform_procs.dm
- code/modules/random_map/noise/desert.dm
- code/modules/random_map/noise/tundra.dm
- code/modules/spells/aoe_turf/summons.dm
- code/modules/xenoarcheaology/artifacts/autocloner.dm
- code/modules/xenoarcheaology/artifacts/replicator.dm
- code/modules/xenoarcheaology/finds/special.dm
- maps/RandomZLevels/wildwest.dm
- maps/southern_cross/southern_cross-6.dmm

And then roughly 3/4ths of all of our PoI maps and the .dme.

File: code/modules/mob/living/simple_mob/subtypes/mechanical/hivebot/hivebot.dm [hunk 1]
```diff
@@ -40,3 +39,11 @@
 	damage_type = BRUTE
 	sharp = FALSE
 	edge = FALSE
+
+/mob/living/simple_mob/mechanical/hivebot/swarm
+	name = "swarm hivebot"
+	desc = "A robot. It looks fragile and weak"
```
Comment (Atermonera) [inline code/modules/mob/living/simple_mob/subtypes/mechanical/hivebot/hivebot.dm]: Missing end punctuation

File: code/modules/mob/living/simple_mob/subtypes/animal/giant_spider/nurse.dm [hunk 2]
```diff
@@ -186,3 +186,9 @@
 	light_range = 2
 	light_power = -3
 
+// This is still stupid, but whatever.
+/mob/living/simple_mob/animal/giant_spider/nurse/hat
+	desc = "Furry and beige, it makes you shudder to look at it. This one has brilliant green eyes and a tiny nurse hat"
```
Comment (Atermonera) [inline code/modules/mob/living/simple_mob/subtypes/animal/giant_spider/nurse.dm]: Missing end punctuation

**Pull request merged**